### PR TITLE
Rework replay requests

### PR DESCRIPTION
**Pull request by lowderchris**
Reworks the replay request cleanup script for automation of data replays.

### CODE REVIEW

**Review by jmbhughes — COMMENTED** [review]
@lowderchris This is the kind of code that would be straightforward and really beneficial to test. Could you write a little scenario where you check the normal case and then some edge cases (empty request, fully connected requests, only one request)? We might also make this a callable script instead of just living in a folder (like we're doing for calibration in punchbowl now).

File: punchpipe/cli.py [hunk 1]
```diff
@@ -238,15 +238,16 @@ def control_process_launcher() -> subprocess.Popen:
                 print("punchpipe abruptly shut down")
 
 
-def clean_replay(input_file: str, write: bool = True, timerange: int = 5) -> None | pd.DataFrame:
+def clean_replay(input_file: str, configuration_path, write: bool = True, timerange: int = 5) -> None | pd.DataFrame:
```
Comment (jmbhughes) [inline punchpipe/cli.py:241]: The timerange could go in the config too. I'd like it to be called something like "window_in_days" though. Just something where the unit is included.

File: config.yaml [hunk 2]
```diff
@@ -16,6 +16,8 @@ ccd_gain:
   3: [4.94, 4.89]
   4: [4.98, 4.92]
 
+science_blocks: [2048, 24575]
```
Comment (jmbhughes) [inline config.yaml:19]: Could we put this in a "replay" subsection?

File: punchpipe/flows/tests/test_replay.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+def test_clean_replay_notime():
+    input_file = os.path.join(TEST_DIR, "data/test_replay.csv")
+    result = clean_replay(input_file, write=False, timerange=None)
+
+    assert isinstance(result, pd.DataFrame)
+    assert len(result) != 0
+
+
+# TODO - other tests - empty request, fully connected requests, only one request
```
Comment (jmbhughes) [inline punchpipe/flows/tests/test_replay.py:27]: Don't forget these tests.

File: punchpipe/flows/tests/test_replay.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+TEST_DIR = os.path.dirname(__file__)
+
+
+def test_clean_replay():
+    input_file = os.path.join(TEST_DIR, "data/test_replay.csv")
+    result = clean_replay(input_file, write=False, timerange=None)
+
+    assert isinstance(result, pd.DataFrame)
+    assert len(result) != 0
```
Comment (jmbhughes) [inline punchpipe/flows/tests/test_replay.py:16]: Do you know the exact number? It would be better to be precise.

File: CHANGELOG.md [hunk 5]
```diff
@@ -7,6 +7,7 @@
 * Fixes metadata usage for quicklook animations in https://github.com/punch-mission/punchpipe/pull/185
 * Relabels CCD halves in https://github.com/punch-mission/punchpipe/pull/174
 * Don't remake existing files in https://github.com/punch-mission/punchpipe/pull/188
+* Updates replay request cleaning script in https://github.com/punch-mission/punchpipe/pull/190
```
Comment (jmbhughes) [inline CHANGELOG.md:10]: Should be in the unreleased section

**Review by jmbhughes — CHANGES_REQUESTED** [review]
Almost there!

**Review by jmbhughes — APPROVED** [review]
